### PR TITLE
EZP-29618: Update user menu order and add icons

### PR DESCRIPTION
**Pull request by tischsoic**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-29618
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

*Related PR: https://github.com/ezsystems/ezplatform-user/pull/46*

# Screenshots
Before             |  After
:-------------------------:|:-------------------------:
Screen Shot 2019-09-04 at 09 55 23  |  Screen Shot 2019-09-04 at 10 44 48

dNrvpPaHof

## Long name

Screen Shot 2019-09-03 at 15 02 19

Screen Shot 2019-09-03 at 14 58 35

## Short name

Screen Shot 2019-09-03 at 15 00 09

Screen Shot 2019-09-03 at 14 58 55

## With notification

Screen Shot 2019-09-03 at 15 11 56

Screen Shot 2019-09-03 at 15 12 13

### CODE REVIEW

**Comment by inakijv** [comment]
Hi @tischsoic Looks cool to me. Because I only created the JIRA ticket you had to guess the alignments. I took a screenshot with them.

Screen Shot 2019-09-03 at 10 01 42 AM

**Comment by tischsoic** [comment]
Fixed styling - ping @inakijv :
Screen Shot 2019-09-04 at 08 45 49

File: src/bundle/Resources/public/scss/_user-menu.scss [hunk 1]
```diff
@@ -1,55 +1,78 @@
...
+            padding-top: calculateRem(6px);
+            top: calc(100% - #{calculateRem(6px)});
             right: 0;
             background-color: $ez-white;
-            box-shadow: 0 2px 4px 0 rgba(0,0,0,.45);
-            height: auto;
+            box-shadow: 0 calculateRem(6px) calculateRem(6px) 0 rgba(51, 51, 51, 0.3);
             transform-origin: top center;
-            transition: all .2s $ez-admin-transition;
             white-space: nowrap;
```
Comment (dew326) [inline src/bundle/Resources/public/scss/_user-menu.scss]:
```suggestion
```

File: src/bundle/Resources/public/scss/_user-menu.scss [hunk 2]
```diff
@@ -58,15 +81,22 @@
...
+                grid-template-areas: 'icon label';
+                padding: calculateRem(4px) calculateRem(4px);
+
                 color: $ez-black;
                 font-size: calculateRem(14px);
-            }
 
-            &:not(:last-child) {
-                border-bottom: 1px solid $ez-ground-base-dark;
+                .ez-icon--small-medium {
```
Comment (dew326) [inline src/bundle/Resources/public/scss/_user-menu.scss]: Duplicated values, to check.

File: src/bundle/Resources/public/scss/_user-menu.scss [hunk 3]
```diff
@@ -58,15 +81,22 @@
...
-            padding: 0 1.5rem;
+            padding: 0 calculateRem(16px);
 
             .nav-link {
+                display: grid;
+                grid-template-columns: calculateRem(20px) 1fr;
+                grid-column-gap: calculateRem(8px);
+                grid-template-areas: 'icon label';
+                padding: calculateRem(4px) calculateRem(4px);
+
```
Comment (dew326) [inline src/bundle/Resources/public/scss/_user-menu.scss]:
```suggestion
```

File: src/bundle/Resources/views/themes/admin/ui/menu/user_menu.html.twig [hunk 4]
```diff
@@ -19,3 +19,17 @@
     {% endif %}
     {{ parent() }}
 {% endblock %}
+
+{% block linkElement %}
+    {% import _self as knp_menu %}
+    <a href="{{ item.uri }}"{{ knp_menu.attributes(item.linkAttributes) }}>
+        {% if item.getExtra('icon') is not null %}
+            <svg class="ez-icon ez-icon--small-medium ez-user-menu__item-icon" aria-hidden="true">
+                <use xlink:href="{{ asset('bundles/ezplatformadminui/img/ez-icons.svg') }}#{{ item.getExtra('icon') }}"></use>
```
Comment (dew326) [inline src/bundle/Resources/views/themes/admin/ui/menu/user_menu.html.twig]: People may want to add custom icon to the list for custom items. I think we should store the whole path (`bundles/ezplatformadminui/img/ez-icons.svg#bookmark-manager`) in the `icon` not only the identifier

Comment (tischsoic) [reply src/bundle/Resources/views/themes/admin/ui/menu/user_menu.html.twig]: Good idea!

File: src/bundle/Resources/public/scss/_user-menu.scss (hunk 2 above)
Comment (tischsoic) [reply src/bundle/Resources/public/scss/_user-menu.scss]: It overwrites this: https://github.com/ezsystems/ezplatform-admin-ui/commit/a323f42fafb589460e636f3ad51e76baaa875773#diff-4a830143277c2bbd9e1d3228c2c048c1R27
But I guess we can delete it?

Comment (dew326) [reply src/bundle/Resources/public/scss/_user-menu.scss]: I think it was before we started to use the modifiers. You can, but you also have to spot all the places where the `--medium` class should be applied.

**Pull request merged**